lib/ansible/plugins/inventory/yc_compute.py

```diff
@@ -0,0 +1,295 @@
+      service_account_contents:
+        description: Similar to service_account_file. Should contain raw contents of the Service Account JSON file.
+        type: string
+        env:
+          - name: YC_ANSIBLE_SERVICE_ACCOUNT_CONTENTS
+        no_log: True
+      hostnames:
+        description:
+          - The list of methods for determining the hostname.
+          - Several methods can be tried one by one. Until successful hostname detection.
+          - Currently supported methods are 'public_ip', 'private_ip' and 'fqdn'.
+          - Any other value is parsed as a jinja2 expression.
+        default: ['public_ip', 'private_ip', 'fqdn']
+        type: list
+      folders:
+        description: List of Yandex.Cloud folder ID's to list instances from.
+        type: list
+        required: True
+      remote_filter:
+        description:
+          - Sets C(filter) parameter for C(list) API call.
+          - Currently you can use filtering only on the Instance.name field.
+          - See U(https://cloud.yandex.com/docs/compute/api-ref/Instance/list).
+          - Use C(filters) option for more flexible client-side filtering.
+        type: string
+      filters:
+        description:
+          - List of jinja2 expressions to perform client-side hosts filtering.
+          - Possible fields are described here U(https://cloud.yandex.com/docs/compute/api-ref/Instance/list).
+          - When overriding this option don't forget to explicitly include default value to your rules (if you need it).
+        type: list
+        default: status == 'RUNNING'
+      api_retry_count:
+        description: Retries count for API calls.
+        type: int
+        default: 5
+'''
+
+EXAMPLES = r'''
+plugin: yc_compute
+folders:  # List inventory hosts from these folders.
+  - <your_folder_id>
+filters:
+  - status == 'RUNNING'
+  - labels['role'] == 'db'
+auth_kind: serviceaccountfile
+service_account_file: /path/to/your/service/account/file.json
+hostnames:
+  - fqdn  # Use FQDN for inventory hostnames.
+# You can also format hostnames with jinja2 epressions like this
+# - "{{id}}_{{name}}"
+
+compose:
+  # Set ansible_host to the Public IP address to connect to the host.
+  # For Private IP use "network_interfaces[0].primary_v4_address.address".
+  ansible_host: network_interfaces[0].primary_v4_address.one_to_one_nat.address
+
+keyed_groups:
+  # Place hosts in groups named by folder_id.
+  - key: folder_id
+    prefix: ''
+    separator: ''
+  # Place hosts in groups named by value of labels['group'].
+  - key: labels['group']
+
+groups:
+  # Place hosts in 'ssd' group if they have appropriate disk_type label.
+  ssd: labels['disk_type'] == 'ssd'
+'''
+
+import json
+
+import grpc
+from yandex.cloud.compute.v1.instance_service_pb2 import ListInstancesRequest
+from yandex.cloud.compute.v1.instance_service_pb2_grpc import InstanceServiceStub
+import yandexcloud
+from google.protobuf.json_format import MessageToDict
```

Comment:
This is minor, but put bare imports first followed by `from` imports.

```suggestion
import json
import grpc
import yandexcloud

from google.protobuf.json_format import MessageToDict 
from yandex.cloud.compute.v1.instance_service_pb2 import ListInstancesRequest
from yandex.cloud.compute.v1.instance_service_pb2_grpc import InstanceServiceStub
```